task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -578,6 +541,19 @@ def run(
     ti: RuntimeTaskInstance, log: Logger
 ) -> tuple[IntermediateTIState | TerminalTIState, ToSupervisor | None, BaseException | None]:
     """Run the task in this process."""
+    # First, clear the xcom data
+    if ti._ti_context_from_server and ti._ti_context_from_server.xcom_keys_to_clear:
+        keys_to_delete = ti._ti_context_from_server.xcom_keys_to_clear
```

Comment:
Nit? (Not sure which is better actually) 
```suggestion
    if ti._ti_context_from_server and (keys_to_delete := ti._ti_context_from_server.xcom_keys_to_clear):
```


However putthingb this above the imports and setup feels wrong. This should be move to inside the try block I think

Reply:
Yeah it is wrong indeed. I will move it right before getting the context and i like your nit better.

Reply:
Handled in 4b183084fc2979a4cb63bd60b74989fefc2e61cd